### ISSUE TITLE
[Bug]: Can't run Shopify app in development using Turbo

### ISSUE THREAD

**Issue by garethlau** [Shopify/cli#1341]
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

App

### Expected behavior

The `turbo run dev` command runs the Shopify app's dev command and should start the backend and frontend.

### Actual behavior

The Shopify app crashes and fails to start.

### Stack trace

```shell
ERROR Raw mode is not supported on the current process.stdin, which Ink uses
as input stream by default.
Read about how to prevent this error on
https://github.com/vadimdemedes/ink/#israwmodesupported
```

### Reproduction steps

1. Create monorepo with turborepo: `npx create-turbo@latest`
2. Create Shopify app in the app directory: `yarn create @shopify/app`
3. Run the dev command: `yarn dev`

### Operating System

Mac OS

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.42.0

### Node version (run `node -v` if you're not sure)

14.19.1

**Comment by Cavallando** [comment]
Just came across this issue today. FWIW I came across this [related issue](https://github.com/vercel/turbo/issues/1235) in `vercel/turbo` which would most likely allow shopify cli to work with turbo since std in would be forwarded to shopify's subprocess instead of consumed by turbo repo.

In addition to vercel adding that feature, IMO it would be nice to have a cli option to run `shopify app dev` in a "headless" mode of sorts that doesn't require those prompts. (probably wouldn't work when `--reset` flag is used in shopify which makes sense).

For now, I was able to get this to work with turbo repo by downgrading both `@shopify/cli` and `@shopify/app` to `3.39.0` which is before the [UI improvements ](https://github.com/Shopify/cli/releases/tag/3.40.0) were added.
